docs/development-guide.md

```diff
@@ -0,0 +1,192 @@
+
+We follow a rough convention for commit messages that is designed to answer two
+questions: what changed and why? The subject line should feature the what and
+the body of the commit should describe the why.
+
+```
+controller: fix bug in configmap
+
+fix clusterID bug in the configmap where the clusterID is
+not set to the expected value.
+
+Signed-off-by: Random J Developer <random@developer.example.org>
+```
+
+The format can be described more formally as follows:
+
+```
+<component>: <subject of the change>
+<BLANK LINE>
+<paragraph(s) with reason/description>
+<BLANK LINE>
+<signed-off-by>
+```
+
+The first line is the subject and should be no longer than 70 characters, the
+second line is always blank, and other lines should be wrapped at 80 characters.
+This allows the message to be easier to read on GitHub as well as in various
+git tools.
+
+Here is a short guide on how to work on a new patch.  In this example, we will
+work on a patch called *hellopatch*:
+
+```console
+git checkout main
+git pull
+git checkout -b hellopatch
+```
+
+Do your work here and commit.
+
+Run the test suite, which includes linting checks, static code check, and unit
+tests:
+
+```console
+make test
+```
+
+Once you are ready to push, you will type the following:
+
+```console
+git push fork hellopatch
+```
+
+**Creating A Pull Request:**
+When you are satisfied with your changes, you will then need to go to your repo
+in GitHub.com and create a pull request for your branch. Automated tests will
+be run against the pull request. Your pull request will be reviewed and merged.
+
+If you are planning on making a large set of changes or a major architectural
+change it is often desirable to first build a consensus in an issue discussion
+and/or create an initial design doc PR. Once the design has been agreed upon
+one or more PRs implementing the plan can be made.
+
+A few labels interact with automation around the pull requests:
+
+* ready-to-merge: This PR is ready to be merged and it doesn't need second review
+* DNM: DO NOT MERGE (Mergify will not merge this PR)
+* ok-to-test: PR is ready for e2e testing.
+
+**Review Process:**
+Once your PR has been submitted for review the following criteria will
+need to be met before it will be merged:
+
+When the criteria are met, a project maintainer can merge your changes into
+the project's main branch.
+
+### Backport a Fix to a Release Branch
+
+The flow for getting a fix into a release branch is:
+
+1. Open a PR to merge the changes to main following the process outlined above.
+2. Add the backport label to that PR such as `backport-to-release-vX.Y.Z`
+3. After your PR is merged to main, the mergify bot will automatically open a
+   PR with your commits backported to the release branch
+4. If there are any conflicts you will need to resolve them by pulling the
+   branch, resolving the conflicts and force push back the branch
+5. After the CI is green, the bot will automatically merge the backport PR.
```

Comment:
Same question as a previous comment: Should we document an automation we don't have yet?

Reply:
no harm in having this and work on the automation in followup. once the PR is merged i will setup the automation for it.

Reply:
ok